NoodleMagazineProvider/src/main/kotlin/com/KillerDogeEmpire/NoodleMagazineProvider.kt

```diff
@@ -0,0 +1,120 @@
+package com.KillerDogeEmpire
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.lagradost.cloudstream3.*
+import com.lagradost.cloudstream3.network.WebViewResolver
+import com.lagradost.cloudstream3.utils.*
+import org.jsoup.nodes.Element
+
+class NoodleMagazineProvider : MainAPI() { // all providers must be an instance of MainAPI
+    override var mainUrl = "https://noodlemagazine.com"
+    override var name = "Noodle Magazine"
+    override val hasMainPage = true
+    override var lang = "en"
+    override val hasDownloadSupport = true
+    override val supportedTypes = setOf(
+            TvType.NSFW
+    )
+
+    override val mainPage = mainPageOf(
+            "latest" to "Latest",
+            "onlyfans" to "Onlyfans",
+            "latina" to "Latina",
+            "blonde" to "Blonde",
+            "milf" to "MILF",
+            "jav" to "JAV",
+            "hentai" to "Hentai",
+            "lesbian" to "Lesbian",
+    )
+
+
+    override suspend fun getMainPage(page: Int, request: MainPageRequest): HomePageResponse {
+        val curpage = page - 1
+        val link = "$mainUrl/video/${request.data}?p=$curpage"
+        val document = app.get(link).document
+        val home = document.select("div.item").mapNotNull {
+            it.toSearchResult()
+        }
+        return newHomePageResponse(request.name, home)
+    }
+
+    private fun Element.toSearchResult(): AnimeSearchResponse? {
+        val href = fixUrl(this.selectFirst("a")?.attr("href") ?: return null)
+        val title = this.selectFirst("a div.i_info div.title")?.text() ?: return null
+        val posterUrl = fixUrlNull(this.selectFirst("a div.i_img img")?.attr("data-src"))
+
+        return newAnimeSearchResponse(title, href, TvType.Anime) {
+            this.posterUrl = posterUrl
+        }
+    }
+
+    override suspend fun search(query: String): List<AnimeSearchResponse> {
+        val searchresult = mutableListOf<AnimeSearchResponse>()
+        (0..10).toList().apmap { page ->
```

Comment:
I don't get this. Why does it needs to iterate at exactly 0-10 range ?

Reply:
Also, change response type. Its not technically Anime so it might be possible it can be confused as such. Use ``newMovieSearchResponse`` instead.

Reply:
Changed it to new moviesearchresponse, it doesn't have to be 0 - 10, however it just seems to work better when set to that. It does get everything though.

Reply:
My main gripe with it is overloading the host with unnecessary requests. It seems bad. It'd be better to just request a single page, and use paginated view that's already done on main App

Reply:
I see, will try and do that tomorrow